### ISSUE TITLE
Experiment #005: Repeat Experiment #003 with duplicates

### ISSUE THREAD

**Issue by AyronK** [AyronK/softmax-dnn-example#8]
Repeat the #10, but with more cards in deck and allow duplicates indicated by value of 2.

## Assumptions
1. Full deck has 5 card slots
2. There is 10 total cards in the pool.
3. Cards are values of one in 10-length array:
   [1,0,0,0,0,0,0,0,0,0] - first card
   [1,0,1,0,1,0,0,0,0,0] - deck containing first, third and fifth card
4. Data is correlated:
```
                // Class 1 decks
                {2, 1, 0, 0, 2, 0, 0, 0, 0, 0},  
                {1, 2, 0, 0, 0, 2, 0, 0, 0, 0},  
                {2, 1, 0, 0, 0, 0, 2, 0, 0, 0},  
                {1, 2, 0, 0, 0, 0, 0, 2, 0, 0},  
                // Class 2 decks  
                {0, 0, 2, 1, 0, 0, 2, 0, 0, 0},  
                {0, 0, 1, 2, 0, 0, 0, 2, 0, 0},  
                {0, 0, 2, 1, 0, 0, 0, 0, 2, 0},  
                {0, 0, 1, 2, 0, 0, 0, 0, 0, 2},  
```
Same rules for card usage.

## Process
1. For each deck remove one "card" and take two as input and one as output (if there is a duplicate we only use this case once - this means we will have 3 inputs for one deck)
    - input:   [2,0,0,0,2,0,0,0,0,0] 
    - output: [0,0,1,0,0,0,0,0,0,0] 
    - **OR for duplicates**
    - input:   [1,0,1,0,2,0,0,0,0,0] 
    - output: [1,0,0,0,0,0,0,0,0,0] 
2. Shuffle the data of 24 total (_8 decks * 3 cards_) options
3. Take 16 options as learning set and 8 as testing set
4. Teach network
5. Test accuracy
6. Test how duplicates affect on the results

**Comment by AyronK** [comment]
## Result
Below the comparison between avarage of 100 tries accuracy for two different networks

### Activation Network
- 5 hidden neurons
- LearningRate = 0.07,
- Momentum = 0.3

**Learing set accuracy:** 99,06%
**Testing set accuracy:** 12,33%

### Deep Belief Neural Network
- 2 hidden layers, 5 neurons each
- Unsupervised
   * LearningRate = 0.07,
   * Momentum = 0.3
- Supervised
   * LearningRate = 0.07,
   * Momentum = 0.2

**Learing set accuracy:** 96,94%
**Testing set accuracy:** 14,75%

**Comment by AyronK** [comment]
## Conclusion

- Accuracy in all tests is far from usefull

I also tested for cases:
- One card: 0.5, Two cards: 1
- One card: 1, Two cards: -1

Results were very close to this experiment in range of 10-20%.

It might be fault of bad data correlation. It should resolve after #9 .
